packages/firestore/src/model/values.ts

```diff
@@ -539,3 +539,43 @@ export function normalizeByteString(blob: string | Uint8Array): ByteString {
     return ByteString.fromUint8Array(blob);
   }
 }
+
+/** Returns true if `value` is an IntegerValue. */
+export function isInteger(value?: api.Value | null): boolean {
```

Comment:
If we defined an interface like

```ts
exporter interface IntegerValue {
  integerValue: number;
}
```

Then we could make this a type guard:

```ts
export function isInteger(value?: api.Value | null): value is IntegerValue {
  return !!value && 'integerValue' in value;
}
```

This PR doesn't include any uses of these functions, so it's not clear that this would necessarily be a win, but it seems like this could be helpful.

Reply:
I updated the PR to allow us to make asserts on the input type, but I kept it in the method declarations for now and did not plumb through separate types. This should allow us to get rid of bangs for future usages.